testdata/project-v4-declarative-v1/monitoring/tools/metricsdocs.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+
+	"sigs.k8s.io/kubebuilder/testdata/project-v4-declarative-v1/monitoring/metrics"
+	"sigs.k8s.io/kubebuilder/testdata/project-v4-declarative-v1/monitoring/metrics/util"
+)
+
+// constant parts of the file
+const (
+	title      = "# sigs.k8s.io/kubebuilder/testdata/project-v4-declarative-v1 Operator Metrics\n"
+	background = "This document aims to help users that are not familiar with metrics exposed by the sigs.k8s.io/kubebuilder/testdata/project-v4-declarative-v1 operator.\n" +
+		"All metrics documented here are auto-generated by the utility tool \"monitoring/tools/metricsdocs.go\" and reflects exactly what is being exposed.\n\n"
+
+	KVSpecificMetrics = "## sigs.k8s.io/kubebuilder/testdata/project-v4-declarative-v1 Operator Metrics List\n"
+
+	opening = title +
+		background +
+		KVSpecificMetrics
+
+	// footer
+	footerHeading = "## Developing new metrics\n"
+	footerContent = "After developing new metrics or changing old ones, please run \"make generate-metricsdocs\" to regenerate this document.\n\n" +
+		"If you feel that the new metric doesn't follow these rules, please change \"monitoring/tools/metricsdocs.go\" with your needs.\n"
+
+	footer = footerHeading + footerContent
+)
+
+func main() {
```

Comment:
Is really required to implement a util to do that?
If that is a common good practice so have not tools and etc that does it already?

Reply:
@camilamacedo86  Yes, This is a very useful addition for creating an automated metrics documentation.
It we be automatically updated after every new metric is added.
We plan to also add to it the recording rules, since they are used the same as metrics

Reply:
I think what @camilamacedo86 is trying to point out here is that if every operator project that wants to create/use metrics needs to use this tooling, it might be better to instead have a library/binary with this functionality that can import/download/build and be used instead of scaffolding it in every operator project.

I agree with @camilamacedo86 here that we should probably consider finding a way to make this code reusable by many operator projects without actually scaffolding this code in every operator project.